src/collection-preferences/utils.tsx

```diff
@@ -68,21 +68,12 @@ export const ModalContentLayout = ({ left, right }: ModalContentLayoutProps) =>
   const [breakpoint, ref] = useContainerBreakpoints(['xs']);
   const smallContainer = breakpoint === 'default';
 
-  if (smallContainer) {
-    return (
-      <div ref={ref}>
-        {left && <div>{left}</div>}
-        {right && <div className={clsx(left && styles['second-column-small'])}>{right}</div>}
-      </div>
-    );
-  }
-
   const columns = left && right ? 2 : 1;
   return (
     <div ref={ref}>
-      <InternalColumnLayout columns={columns} variant="text-grid">
+      <InternalColumnLayout columns={smallContainer ? 1 : columns} variant="text-grid">
         {left && <div>{left}</div>}
-        {right && <div>{right}</div>}
+        {right && <div className={clsx(left && smallContainer && styles['second-column-small'])}>{right}</div>}
       </InternalColumnLayout>
```

Comment:
These two return statements were causing the state to be lost in the filtering, updated this section to only have on return statement. When `smallContainer` is true the `columnLayout` is reduced to 1 column to mimic the original layout from the `return` above, it is also used to conditionally apply the relevant class when true